src/definition/populate.go

```diff
@@ -85,7 +85,14 @@ func IntegrationProtocol2PopulateFunc(
 					continue
 				}
 
-				ms := e.NewMetricSet(msType)
```

Comment:
~This won't help with our issue because different jobs calls populate and `IntegrationProtocol2PopulateFunc` multiple times anyway. We would have to globally share the metric sets map by type, which doesn't seem like a good idea.~